src/shipping/components/ShippingZoneRatesCreatePage/ShippingZoneRatesCreatePage.tsx

```diff
@@ -0,0 +1,169 @@
+import { ChannelShippingData } from "@saleor/channels/utils";
+import AppHeader from "@saleor/components/AppHeader";
+import CardSpacer from "@saleor/components/CardSpacer";
+import ChannelsAvailability from "@saleor/components/ChannelsAvailability";
+import { ConfirmButtonTransitionState } from "@saleor/components/ConfirmButton";
+import Container from "@saleor/components/Container";
+import Form from "@saleor/components/Form";
+import Grid from "@saleor/components/Grid";
+import PageHeader from "@saleor/components/PageHeader";
+import SaveButtonBar from "@saleor/components/SaveButtonBar";
+import { ShippingChannelsErrorFragment } from "@saleor/fragments/types/ShippingChannelsErrorFragment";
+import { ShippingErrorFragment } from "@saleor/fragments/types/ShippingErrorFragment";
+import { validatePrice } from "@saleor/products/utils/validation";
+import OrderValue from "@saleor/shipping/components/OrderValue";
+import OrderWeight from "@saleor/shipping/components/OrderWeight";
+import PricingCard from "@saleor/shipping/components/PricingCard";
+import ShippingZoneInfo from "@saleor/shipping/components/ShippingZoneInfo";
+import { createChannelsChangeHandler } from "@saleor/shipping/handlers";
+import { ShippingMethodTypeEnum } from "@saleor/types/globalTypes";
+import React from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+export interface FormData {
+  channelListings: ChannelShippingData[];
+  name: string;
+  noLimits: boolean;
+  minValue: string;
+  maxValue: string;
+  type: ShippingMethodTypeEnum;
+}
+
+export interface ShippingZoneRatesCreatePageProps {
+  allChannelsCount?: number;
+  shippingChannels: ChannelShippingData[];
+  disabled: boolean;
+  hasChannelChanged?: boolean;
+  channelErrors: ShippingChannelsErrorFragment[];
+  errors: ShippingErrorFragment[];
+  saveButtonBarState: ConfirmButtonTransitionState;
+  onBack: () => void;
+  onDelete?: () => void;
+  onSubmit: (data: FormData) => void;
+  onChannelsChange: (data: ChannelShippingData[]) => void;
+  openChannelsModal: () => void;
+  variant: ShippingMethodTypeEnum;
+}
+
+export const ShippingZoneRatesCreatePage: React.FC<ShippingZoneRatesCreatePageProps> = ({
+  allChannelsCount,
+  shippingChannels,
+  channelErrors,
+  disabled,
+  errors,
+  hasChannelChanged,
+  onBack,
+  onDelete,
+  onSubmit,
+  onChannelsChange,
+  openChannelsModal,
+  saveButtonBarState,
+  variant
+}) => {
+  const intl = useIntl();
+  const isPriceVariant = variant === ShippingMethodTypeEnum.PRICE;
+  const initialForm: FormData = {
+    channelListings: shippingChannels,
+    maxValue: "",
+    minValue: "",
+    name: "",
+    noLimits: false,
+    type: null
+  };
+
+  return (
+    <Form initial={initialForm} onSubmit={onSubmit}>
```

Comment:
I think we may start to create distinct form components instead of using this generic one for any new component, like this one. You may check out as an example `ProductUpdatePage` or `ProductCreatePage`, which already has a distinct form implemented.

Reply:
It will be done in another pr